com.unity.sg2/Editor/GraphDeltaRegistry/Registry/GraphType.cs

```diff
@@ -233,25 +374,14 @@ public enum Usage { In, Out, Static, Local }
 
         public void BuildType(FieldHandler field, Registry registry)
         {
-            // default initialize to a float4;
-            field.AddSubField(kPrecision, Precision.Single, true);
-            field.AddSubField(kPrimitive, Primitive.Float, true);
-            field.AddSubField(kLength, Length.Four, true);
-            field.AddSubField(kHeight, Height.One, true);
-
-
-            var lenField = field.GetSubField<Length>(kLength);
-            var hgtField = field.GetSubField<Height>(kHeight);
-            var length = Length.Four;
-            var height = Height.One;
-
-            // read userdata and make sure we have enough fields.
-            if (lenField != null) length = lenField.GetData();
-            if (hgtField != null) height = hgtField.GetData();
-
-            // ensure that enough subfield values exist to represent userdata's current data.
-            // we could just ignore userData though and just fill out all 16 possible components...
-            for (int i = 0; i < (int)length * (int)height; ++i)
+            // TODO: Default initialization should be a non-dynamic scalar single precision float.
+            field.AddSubField(kPrecision, Precision.Single, true).AddSubField(kDynamic, false, true);
+            field.AddSubField(kPrimitive, Primitive.Float, true).AddSubField(kDynamic, false, true);
+            field.AddSubField(kLength, Length.One, true).AddSubField(kDynamic, false, true);
+            field.AddSubField(kHeight, Height.One, true).AddSubField(kDynamic, false, true);
+
+            // ensure we have enough allocated.
+            for (int i = 0; i < 16; ++i)
                 GraphTypeHelpers.SetComponent(field, i, 0);
         }
```

Comment:
Just made it so we always initialize 16 floats and also add Dynamic subfields to all the dynamic fields.